#!/usr/bin/env python
PACKAGE = "dafan_docking"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("ang_gains_init_p",    int_t,    0, "An Integer parameter", 50,  0, 100)
gen.add("ang_gains_init_p", double_t, 0, "P for initial angular gain", 0.1,  0.0,  10.0)
gen.add("ang_gains_init_i", double_t, 0, "I for initial angular gain", 0.0,  0.0,  10.0)
gen.add("ang_gains_init_d", double_t, 0, "D for initial angular gain", 0.0,  0.0,  10.0)

gen.add("ang_gains_final_p", double_t, 0, "P for final angular gain", 0.1,  0.0,  10.0)
gen.add("ang_gains_final_i", double_t, 0, "I for final angular gain", 0.0,  0.0,  10.0)
gen.add("ang_gains_final_d", double_t, 0, "D for final angular gain", 0.0,  0.0,  10.0)

gen.add("dist_gains_p", double_t, 0, "P for dist gain", 0.1,  0.0,  10.0)
gen.add("dist_gains_i", double_t, 0, "I for dist gain", 0.0,  0.0,  10.0)
gen.add("dist_gains_d", double_t, 0, "D for dist gain", 0.0,  0.0,  10.0)

gen.add("angle_tol", double_t, 0, "Angular tolerance for goal", 0.0436332 ,  0.0,  0.5)
gen.add("dist_tol", double_t, 0, "Distance tolerance for goal", 0.01,  0.0,  0.1)

exit(gen.generate(PACKAGE, "dafan_docking", "planner_adjuster"))